gubernator/main.py

```diff
@@ -158,24 +167,61 @@ def build_details(build_dir):
         return
     started = json.loads(started)
     finished = json.loads(finished)
+
     failures = []
+    junit_futures = {}
     junit_paths = [f.filename for f in gcs_ls('%s/artifacts' % build_dir)
                    if re.match(r'junit_.*\.xml', os.path.basename(f.filename))]
-    junit_futures = [gcs_async.read(f) for f in junit_paths]
+    
+    fps = []
+    last_len = 0
+    total_len = 0
+    for f in junit_paths:
+        junit_futures[gcs_async.read(f)] = f
+
+
     for future in junit_futures:
         junit = future.get_result()
         if junit is None:
             continue
         failures.extend(parse_junit(junit))
+        total_len = len(fps)
+        last_len = len(failures) - total_len
+        for i in xrange(last_len):
+            fps.append(junit_futures[future])
+
     build_log = None
     if finished and finished.get('result') != 'SUCCESS' and len(failures) == 0:
         build_log = gcs_async.read(build_dir + '/build-log.txt').get_result()
         if build_log:
             build_log = log_parser.digest(build_log.decode('utf8', 'replace'))
             logging.info('fallback log parser emitted %d lines',
                          build_log.count('\n'))
-    return started, finished, failures, build_log
-
+    return started, finished, failures, build_log, fps
```

Comment:
You'll need to change the docstring at the top of the function to reflect the changed return values.